### PR TITLE
[CSRSRV] CsrUnhandledExceptionFilter: Check RtlAdjustPrivilege() result

### PR DESCRIPTION
**Pull request by SergeGautherie**
And remove unused NtRaiseHardError() assignment.

Detected by Cppcheck: redundantAssignment.
Addendum to d221bdfbb1581227c0933e7cf93c7de9cdead01d (r55647).

### CODE REVIEW

File: subsystems/win32/csrsrv/server.c [hunk 1]
```diff
@@ -153,12 +153,11 @@ CsrLoadServerDll(IN PCHAR DllString,
...
 
             /* Send a hard error */
-            NtRaiseHardError(Status,
-                             2,
-                             3,
-                             Parameters,
-                             OptionOk,
-                             &Response);
+            Status2 = NtRaiseHardError(Status, 2, 3, Parameters, OptionOk, &Response);
+            if (!NT_SUCCESS(Status2))
```
Comment (HBelusca) [inline subsystems/win32/csrsrv/server.c:157]: It doesn't really make sense to look for the returned status of NtRaiseHardError, if it's just for dprinting stuff in the debug log. The idea here was to just raise the hard-error, and then the code will continue execution.

File: subsystems/win32/csrsrv/server.c [hunk 2]
```diff
@@ -676,8 +679,13 @@ CsrUnhandledExceptionFilter(IN PEXCEPTION_POINTERS ExceptionInfo)
                                       ErrorParameters,
                                       OptionShutdownSystem,
                                       &Response);
+            if (!NT_SUCCESS(Status))
```
Comment (HBelusca) [inline subsystems/win32/csrsrv/server.c:682]: What for? The CSRSS is killed just afterwards, so there is no real point in telling why NtRaiseHardError could have failed. (It's not like if we wanted to then do actions depending on the returned status.)

File: subsystems/win32/csrsrv/server.c [hunk 3]
```diff
@@ -153,12 +153,11 @@ CsrLoadServerDll(IN PCHAR DllString,
             RtlInitUnicodeString(&ErrorString, L"Default Load Path");
 
             /* Send a hard error */
-            NtRaiseHardError(Status,
-                             2,
-                             3,
-                             Parameters,
-                             OptionOk,
-                             &Response);
+            Status2 = NtRaiseHardError(Status, 2, 3, Parameters, OptionOk, &Response);
```
Comment (HBelusca) [inline subsystems/win32/csrsrv/server.c:156]: Please restore the formatting (and we don't really need the returned Status code, see below).

Comment (SergeGautherie) [reply subsystems/win32/csrsrv/server.c:156]: Reverted.

File: subsystems/win32/csrsrv/server.c (hunk 1 above)
Comment (SergeGautherie) [reply subsystems/win32/csrsrv/server.c:157]: Reverted, then.

File: subsystems/win32/csrsrv/server.c (hunk 2 above)
Comment (SergeGautherie) [reply subsystems/win32/csrsrv/server.c:682]: Reverted, then.

**Pull request merged**